api/src/main/java/org/openmrs/module/insuranceclaims/api/dao/impl/InsuranceClaimDaoImpl.java

```diff
@@ -0,0 +1,39 @@
+package org.openmrs.module.insuranceclaims.api.dao.impl;
+
+import org.hibernate.criterion.Restrictions;
+import org.openmrs.api.db.hibernate.DbSession;
+import org.openmrs.api.db.hibernate.DbSessionFactory;
+import org.openmrs.module.insuranceclaims.api.dao.InsuranceClaimDao;
+import org.openmrs.module.insuranceclaims.api.model.InsuranceClaim;
+import org.openmrs.module.insuranceclaims.util.InsuranceClaimsConstants;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository("insuranceclaims.InsuranceClaimDao")
+public class InsuranceClaimDaoImpl implements InsuranceClaimDao {
+
+	@Autowired
+	private DbSessionFactory sessionFactory;
+
+	public DbSession getSession() {
+		return sessionFactory.getCurrentSession();
+	}
+
+	@Override
```

Comment:
try to make implementation more generic use T

Reply:
Thanks. I will extend the HibernateOpenmrsDataDAO class.